geopandas/io/file.py

```diff
@@ -430,20 +437,40 @@ def _read_file_pyogrio(path_or_bytes, bbox=None, mask=None, rows=None, **kwargs)
             raise TypeError("'rows' must be an integer or a slice.")
     if bbox is not None:
         if isinstance(bbox, (GeoDataFrame, GeoSeries)):
-            bbox = tuple(bbox.total_bounds)
+            crs = pyogrio.read_info(path_or_bytes).get("crs")
+            if isinstance(path_or_bytes, IOBase):
+                path_or_bytes.seek(0)
+
+            bbox = tuple(bbox.to_crs(crs).total_bounds)
         elif isinstance(bbox, BaseGeometry):
             bbox = bbox.bounds
         if len(bbox) != 4:
             raise ValueError("'bbox' should be a length-4 tuple.")
-    if mask is not None:
-        raise ValueError(
-            "The 'mask' keyword is not supported with the 'pyogrio' engine. "
-            "You can use 'bbox' instead."
-        )
+    elif mask is not None:
```

Comment:
should we check and raise if both mask and bbox are provided, currently mask would be silently ignored?
(I'm not sure if we do these kind of checks otherwise / if we check it in the fiona engine, just checking at a glance)

Reply:
Fiona already checks for this, as does pyogrio.  But since we were using an `elif` here between them, I went ahead and added a check on the GeoPandas side that returns the same error message as Fiona.